src/Compilers/CSharp/Portable/Symbols/TypeParameterSymbol.cs

```diff
@@ -690,10 +690,13 @@ internal override TypeSymbol MergeNullability(TypeSymbol other, VarianceKind var
 
         #region ITypeParameterTypeSymbol Members
 
-        CodeAnalysis.NullableAnnotation ITypeParameterSymbol.ReferenceTypeConstraintNullableAnnotation
+        CodeAnalysis.NullableAnnotation ITypeParameterSymbol.ReferenceTypeConstraintNullableAnnotation => ReferenceTypeConstraintIsNullable switch
         {
-            get => default;
-        }
+            false when !HasReferenceTypeConstraint => CodeAnalysis.NullableAnnotation.NotApplicable,
```

Comment:
>!HasReferenceTypeConstraint [](start = 23, length = 27)

One ```false``` case with conditional inside might be easier to follow.

Reply:
I felt that this was clear and easy to follow.

---
In reply to: [270178845](https://github.com/dotnet/roslyn/pull/34429#discussion_r270178845) [](ancestors = 270178845)